mpags-cipher.cpp

```diff
@@ -0,0 +1,142 @@
+#include <iostream>
+#include <vector>
+#include <string>
+
+int main(int argc, char* argv[]){
+
+    // Convert command line arguments into an easier to use form
+    const std::vector<std::string> cmdLineArgs{argv, argv+argc};
+    const std::size_t nCmdLineArgs(cmdLineArgs.size());
+
+    // Help requested boolean and message
+    bool helpRequested{false};
+    std::string help_message{
+        """Usage: ./mpags-cipher [options]\n"""
+        """Options:\n"""
+        """ -h, --help: Display this message\n"""
+        """ --version: Display version\n"""
+        """ -i FILE  ,  --input-file FILE: Specify the input file\n"""
+        """ -o FILE  ,  --output-file FILE: Specify the output file"""
+        };
+
+    // Versioning
+    bool versionRequested{false};
+    const std::string version_number{"0.1.0"};
+
+    // Input file
+    std::string input_file{""};
+    bool input_file_given{false};
+
+    // Output file
+    std::string output_file{""};
+    bool output_file_given{false};
+
+    for (size_t i{1}; i < nCmdLineArgs; i++){
+        if (cmdLineArgs[i] == "--help" || cmdLineArgs[i] == "-h"){
+            helpRequested = true;
+        }
+        else if (cmdLineArgs[i] == "--version"){
+            versionRequested = true;
+        }
+        else if (cmdLineArgs[i] == "--input-file" || cmdLineArgs[i] == "-i"){
+            // Check if the list is long enough
+            if (i < nCmdLineArgs - 1){
+                input_file = cmdLineArgs[i+1];
+                input_file_given = true;
+                i++; // Increment so that the argument isn't reparsed
+            }
+            else{
+                std::cout << "Error: No input file specified" << std::endl;
```

Comment:
When printing errors or warnings it is usual to use `std::cerr` instead of `std::cout` to allow the output and errors to be written to separate streams if the user desires.